### DISCUSSION TITLE
Why is the connection not guaranteed to be valid?

### DISCUSSION THREAD

**Discussion by jorge-lavin** [xetorthio/jedis#2376] [Q&A]
About this code snippet:

https://github.com/xetorthio/jedis/blob/72e5b1c6184047ba746de0e28de2b788adc898b9/src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java#L64-L81

- Why is the connection not guaranteed to be valid?
- How to obtain a valid connection?

#### Jedis version:
2.9.3

#### Java version:
1.8.151

**Comment by gkorland** [comment]
> * Why is the connection not guaranteed to be valid?
Cause slots might moved.

> * How to obtain a valid connection?
You can't since it's a distributed system, you can test the connection after you get and if the server doesn't return "MOVED" you are ok
